sympy/polys/polyroots.py

```diff
@@ -504,33 +504,42 @@ def roots_cyclotomic(f, factor=False):
 
 def roots_quintic(f):
     """
-    Calculate exact roots of a solvable quintic
+    Calculate exact roots of a solvable irreducible quintic with rational coefficients.
+    Return an empty list if the quintic is reducible or not solvable.
     """
     result = []
-    coeff_5, coeff_4, p, q, r, s = f.all_coeffs()
-
-    # Eqn must be of the form x^5 + px^3 + qx^2 + rx + s
-    if coeff_4:
+    if not f.is_univariate:
         return result
```

Comment:
Whether to check things like this raises the usual questions around public vs private functions, and GIGO.

The name of `roots_quintic()` does not begin with a `_`, which tends to make it look like a public function, but I don't find this function in the API reference either, which tends to make it look private. I don't know if we've settled the policy on this yet -- see #23037.

I suspect that up to now `roots_quintic()` has largely been used only by the (explicitly public) `roots()` function, and that this is why the `r2 is not None` check below was not caught before now.

Is there a reason to call `roots_quintic()` directly, instead of just calling `roots()`?

If not, then maybe checks like these aren't necessary. If so, then the function should have a better docstring, and maybe even more such checks, like checking that `f` is actually degree 5, so that the `coeff_5, coeff_4, p_, q_, r_, s_ = f.all_coeffs()` line can't fail with a `ValueError`.

Reply:
You are right. I think I shall remove this line. Other functions like `roots_cubic` does not check this either.